### PR TITLE
Setting Up Initial React App Skeleton with Basic Routing

### PR DESCRIPTION
**Pull request by abbiehuynh**
## PR includes:
- creating a new React app using Vite
- organized project structure of components and pages
- implementing basic routing using ‘react-router-dom’
  - configured routes for Home, Favorites, and Profile pages
- adding a navigation bar for routing between pages
- adding simple components to verify that app is running correctly
- update: added routing for all pages and organized component structure

### Testing:
- tested the connection between the server and client
  - passes the message from the server and renders through the client onto the webpage in App.jsx file.

## Testing Instructions:
To test the client, clone the repository
```
https://github.com/abbiehuynh/my-crochet-corner.git <NameNewDirectory>
```
and run the following commands in the server:
1. `npm install`
2. `npm start`

and run the following commands in the client:
1. `npm install`
2. `npm run dev`
4. Open http://localhost:5173 in your browser
- test routing to see if each link renders a div with the name of the link.
  - example: Profile link should render “Profile”
- update: test routing for user flow
User Flow

### Notes:
Please review the app structure and suggest any improvements.

---
Merging client branch into app-setup branch

### CODE REVIEW

**Comment by stmcpeters** [comment]
hey abbie! your file structure is so organized, clearly separating the client and server side. I love how you structured your client-side files! it's so smart to have your components organized by the respective pages you're going to route to. I also tried all your routes and they each go to the intended page/component. i honestly have no notes, you've done such amazing work so far. i'm excited to see your next steps! :D

**Comment by abbiehuynh** [comment]
It was my first time organizing the components by their respective pages, so it is good to hear that you were able to see the relationship structure while reviewing them. Thank you so much for testing the user flow and for your feedback, Steph. I look forward to collaborating with you more!

File: client/src/components/home/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    <ListProjects />
+
+    <Link to={`/ai-chat`}>
+        <Button>Open AI</Button>
+    </Link>
+
+    <Link to={`/add-project`}>
+        <Button>Add New Project</Button>
+    </Link>
```
Comment (davidmedina-contact) [inline client/src/components/home/Home.jsx]: Nit: Check the indentation of all these inside the <div>

File: client/.vite/deps_temp_7c811d89/package.json
Comment (davidmedina-contact) [inline client/.vite/deps_temp_7c811d89/package.json:1]: Not sure how Vite works but is this file supposed to be committed?

File: client/src/components/NavBar.jsx
Comment (davidmedina-contact) [inline client/src/components/NavBar.jsx:1]: For organization, maybe NavBar should go in a folder called "common"?

**Review by davidmedina-contact — APPROVED** [review]
Good progress! Minor things related to organization. Looking forward to see more

File: client/src/components/home/Home.jsx (hunk 1 above)
Comment (abbiehuynh) [reply client/src/components/home/Home.jsx]: That is a good call out, thank you for noting this!

File: client/.vite/deps_temp_7c811d89/package.json
Comment (abbiehuynh) [reply client/.vite/deps_temp_7c811d89/package.json:1]: This was something that just showed up after importing something, but I will look more into it. If it is not, then I will move it to .gitignore!

File: client/src/components/NavBar.jsx
Comment (abbiehuynh) [reply client/src/components/NavBar.jsx:1]: That is a good idea, I will add a spot for it in the component's folder.

Comment (abbiehuynh) [reply client/src/components/NavBar.jsx:1]: Are there any other component/features that come to mind that you would suggest should also go into the "common" folder?

Comment (davidmedina-contact) [reply client/src/components/NavBar.jsx:1]: Anything that is part of the navigation, layout, or UI elements present in all or most pages

**Pull request merged**